cove/lib/common.py

```diff
@@ -1,26 +1,280 @@
+
+            try:
+                extension = requests.get(url)
+            except requests.exceptions.RequestException:
+                self.invalid_extension[extensions_descriptor_url] = 'fetching failed'
+                continue
+            if extension.ok:
+                try:
+                    extension_data = extension.json()
+                except json.JSONDecodeError:
+                    self.invalid_extension[extensions_descriptor_url] = 'invalid JSON'
+                    continue
+            else:
+                self.invalid_extension[extensions_descriptor_url] = '{}: {}'.format(extension.status_code,
+                                                                            extension.reason.lower())
+                continue
+
+            schema_obj = json_merge_patch.merge(schema_obj, extension_data)
+            extensions_descriptor = requests.get(extensions_descriptor_url).json()
+            self.extensions[extensions_descriptor_url] = (url, extensions_descriptor['name'],
+                                                          extensions_descriptor['description'])
+            self.extended = True
+
+    def get_release_schema_obj(self, deref=False):
+        release_schema_obj = self._release_schema_obj
+        if self.extended_schema_file:
+            with open(self.extended_schema_file) as fp:
+                release_schema_obj = json.load(fp)
+        elif self.extensions:
+            release_schema_obj = deepcopy(self._release_schema_obj)
+            self.apply_extensions(release_schema_obj)
+        if deref:
+            if self.extended:
+                extended_release_schema_str = json.dumps(release_schema_obj)
+                release_schema_obj = self.deref_schema(extended_release_schema_str)
+            else:
+                release_schema_obj = self.deref_schema(self.release_schema_str)
+        return release_schema_obj
+
+    def get_release_pkg_schema_obj(self, deref=False):
+        package_schema_obj = self._release_pkg_schema_obj
+        if deref:
+            deref_release_schema_obj = self.get_release_schema_obj(deref=True)
+            if self.extended:
+                package_schema_obj = deepcopy(self._release_pkg_schema_obj)
+                package_schema_obj['properties']['releases']['items'] = {}
+                release_pkg_schema_str = json.dumps(package_schema_obj)
+                package_schema_obj = self.deref_schema(release_pkg_schema_str)
+                package_schema_obj['properties']['releases']['items'].update(deref_release_schema_obj)
+            else:
+                package_schema_obj = self.deref_schema(self.release_pkg_schema_str)
+        return package_schema_obj
+
+    def create_extended_release_schema_file(self, upload_dir, upload_url):
+        filepath = os.path.join(upload_dir, 'extended_release_schema.json')
+        if not self.extended or os.path.exists(filepath):
+            return
```

Comment:
Returning here means that:
* If I refresh the page `schema_ocds.extended_schema_url` is not set, so the link in the template is broken.
* The extended schema file is not updated if I change the version.